lib/plugins/aws/deploy/compile/functions/index.js

```diff
@@ -17,15 +17,43 @@ class AwsCompileFunctions {
     };
   }
 
-  compileRole(role) {
-    if (typeof role === 'object') {
-      // role is an "Fn::GetAtt" object
-      return role;
-    } else if (role.indexOf(':') === -1) {
-      // role is a Logical Role Name
-      return { 'Fn::GetAtt': [role, 'Arn'] };
+  compileRole(newFunction, role) {
+    const compiledFunction = newFunction;
+    const unnsupportedRole = new Error(`Unsupported role provided: "${role}"`);
```

Comment:
This should be a Serverless Error (`this.serverless.classes.Error('foo')`) so that the CLI will print it in a Serverless CLI context. (see usage e.g. here: https://github.com/serverless/serverless/blob/f817933909cb524dcde729348bf229ae355e15ac/lib/plugins/package/lib/validate.js#L8-L9)